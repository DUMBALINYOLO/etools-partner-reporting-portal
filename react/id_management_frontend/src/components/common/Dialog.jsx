import React, {Component} from 'react';
import {AppBar, IconButton, Typography, Dialog as MDialog, DialogContent, Toolbar} from "@material-ui/core/";
import Close from "@material-ui/icons/Close";
import {withStyles} from "@material-ui/core/styles";
import PropTypes from "prop-types";

const styleSheet = (theme) => ({
    appBar: {
        boxShadow: 'none'
    },
    toolbar: {
        display: 'flex',
        justifyContent: 'space-between'
    },
    close: {
        color: theme.palette.common.whiteAlpha,
        marginRight: -14
    },
    content: {
        padding: theme.spacing.unit * 3
    },
    caption: {
        padding: `${theme.spacing.unit}px ${theme.spacing.unit * 3}px`,
        backgroundColor: theme.palette.common.grey,
    },
    dialogLoading: {
        pointerEvents: 'none',
    }
});

class Dialog extends Component {
    render() {
        const {open, onClose, title, children, classes, caption, loading} = this.props;

        return (
            <MDialog
                open={!!open}
                onClose={onClose}
                fullWidth
                disableBackdropClick={loading}
                disableEscapeKeyDown={loading}
                className={loading ? classes.dialogLoading : ''}
            >
                <AppBar position="static" className={classes.appBar}>
                    <Toolbar className={classes.toolbar}>
                        <Typography variant="title" color="inherit">
                            {title}
                        </Typography>
                        {!loading && <IconButton className={classes.close} onClick={onClose} aria-label="Close">
                            <Close/>
                        </IconButton>}
                    </Toolbar>
                </AppBar>
                {caption && <div className={classes.caption}>
                    <Typography variant="caption">
                        {caption}
                    </Typography>
                </div>}
                <DialogContent className={classes.content}>
                    {children}
                </DialogContent>
            </MDialog>
        );
    }
}

Dialog.propTypes = {
    caption: PropTypes.string,
    children: PropTypes.any.isRequired,
    classes: PropTypes.object.isRequired,
    loading: PropTypes.bool,
    onClose: PropTypes.func.isRequired,
    open: PropTypes.bool,
    title: PropTypes.string.isRequired
};

export default withStyles(styleSheet)(Dialog);