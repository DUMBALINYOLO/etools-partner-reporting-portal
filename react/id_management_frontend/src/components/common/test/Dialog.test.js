import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import Dialog from '../Dialog';

describe('Dialog component', () => {
    it('renders component correctly and calls click', () => {
        const children = {};
        const classes = {};
        const onClose = jest.fn();
        const title = 'hello';
        const loading = true;
        const caption = 'yay';

        const wrapper = shallow(<Dialog
            onClose={onClose}
            children={children}
            classes={classes}
            title={title}
            loading={loading}
            caption={caption}
        />);

        expect(toJSON(wrapper)).toMatchSnapshot();
        expect(wrapper.dive().length).toBe(1);
    });
});