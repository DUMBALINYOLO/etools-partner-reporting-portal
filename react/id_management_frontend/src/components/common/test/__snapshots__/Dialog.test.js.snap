// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component renders component correctly and calls click 1`] = `
<WithStyles(Dialog)
  caption="yay"
  classes={
    Object {
      "close": "iconButton",
      "dialogLoading": "dialog",
    }
  }
  loading={true}
  onClose={[Function]}
  title="hello"
>
  <Dialog
    caption="yay"
    classes={
      Object {
        "appBar": "Dialog-appBar-1",
        "caption": "Dialog-caption-5",
        "close": "Dialog-close-3 iconButton",
        "content": "Dialog-content-4",
        "dialogLoading": "Dialog-dialogLoading-6 dialog",
        "toolbar": "Dialog-toolbar-2",
      }
    }
    loading={true}
    onClose={[Function]}
    title="hello"
  >
    <WithStyles(Dialog)
      className="Dialog-dialogLoading-6 dialog"
      disableBackdropClick={true}
      disableEscapeKeyDown={true}
      fullWidth={true}
      maxWidth="sm"
      onClose={[Function]}
      open={false}
    >
      <Dialog
        TransitionComponent={[Function]}
        className="Dialog-dialogLoading-6 dialog"
        classes={
          Object {
            "paper": "MuiDialog-paper-10",
            "paperFullScreen": "MuiDialog-paperFullScreen-17",
            "paperFullWidth": "MuiDialog-paperFullWidth-16",
            "paperScrollBody": "MuiDialog-paperScrollBody-12",
            "paperScrollPaper": "MuiDialog-paperScrollPaper-11",
            "paperWidthMd": "MuiDialog-paperWidthMd-15",
            "paperWidthSm": "MuiDialog-paperWidthSm-14",
            "paperWidthXs": "MuiDialog-paperWidthXs-13",
            "root": "MuiDialog-root-7",
            "scrollBody": "MuiDialog-scrollBody-9",
            "scrollPaper": "MuiDialog-scrollPaper-8",
          }
        }
        disableBackdropClick={true}
        disableEscapeKeyDown={true}
        fullScreen={false}
        fullWidth={true}
        maxWidth="sm"
        onClose={[Function]}
        open={false}
        scroll="paper"
        transitionDuration={
          Object {
            "enter": 225,
            "exit": 195,
          }
        }
      >
        <WithStyles(Modal)
          BackdropProps={
            Object {
              "transitionDuration": Object {
                "enter": 225,
                "exit": 195,
              },
            }
          }
          className="MuiDialog-root-7 MuiDialog-scrollPaper-8 Dialog-dialogLoading-6 dialog"
          disableBackdropClick={true}
          disableEscapeKeyDown={true}
          onClose={[Function]}
          open={false}
          role="dialog"
        >
          <Modal
            BackdropComponent={[Function]}
            BackdropProps={
              Object {
                "transitionDuration": Object {
                  "enter": 225,
                  "exit": 195,
                },
              }
            }
            className="MuiDialog-root-7 MuiDialog-scrollPaper-8 Dialog-dialogLoading-6 dialog"
            classes={
              Object {
                "hidden": "MuiModal-hidden-19",
                "root": "MuiModal-root-18",
              }
            }
            disableAutoFocus={false}
            disableBackdropClick={true}
            disableEnforceFocus={false}
            disableEscapeKeyDown={true}
            disablePortal={false}
            disableRestoreFocus={false}
            hideBackdrop={false}
            keepMounted={false}
            manager={
              ModalManager {
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "modals": Array [],
              }
            }
            onClose={[Function]}
            open={false}
            role="dialog"
          />
        </WithStyles(Modal)>
      </Dialog>
    </WithStyles(Dialog)>
  </Dialog>
</WithStyles(Dialog)>
`;

exports[`Dialog component renders properly when loading is false 1`] = `
<WithStyles(Dialog)
  caption="yay"
  classes={
    Object {
      "close": "iconButton",
      "dialogLoading": "dialog",
    }
  }
  loading={false}
  onClose={[Function]}
  title="hello"
>
  <Dialog
    caption="yay"
    classes={
      Object {
        "appBar": "Dialog-appBar-1",
        "caption": "Dialog-caption-5",
        "close": "Dialog-close-3 iconButton",
        "content": "Dialog-content-4",
        "dialogLoading": "Dialog-dialogLoading-6 dialog",
        "toolbar": "Dialog-toolbar-2",
      }
    }
    loading={false}
    onClose={[Function]}
    title="hello"
  >
    <WithStyles(Dialog)
      className=""
      disableBackdropClick={false}
      disableEscapeKeyDown={false}
      fullWidth={true}
      maxWidth="sm"
      onClose={[Function]}
      open={false}
    >
      <Dialog
        TransitionComponent={[Function]}
        className=""
        classes={
          Object {
            "paper": "MuiDialog-paper-10",
            "paperFullScreen": "MuiDialog-paperFullScreen-17",
            "paperFullWidth": "MuiDialog-paperFullWidth-16",
            "paperScrollBody": "MuiDialog-paperScrollBody-12",
            "paperScrollPaper": "MuiDialog-paperScrollPaper-11",
            "paperWidthMd": "MuiDialog-paperWidthMd-15",
            "paperWidthSm": "MuiDialog-paperWidthSm-14",
            "paperWidthXs": "MuiDialog-paperWidthXs-13",
            "root": "MuiDialog-root-7",
            "scrollBody": "MuiDialog-scrollBody-9",
            "scrollPaper": "MuiDialog-scrollPaper-8",
          }
        }
        disableBackdropClick={false}
        disableEscapeKeyDown={false}
        fullScreen={false}
        fullWidth={true}
        maxWidth="sm"
        onClose={[Function]}
        open={false}
        scroll="paper"
        transitionDuration={
          Object {
            "enter": 225,
            "exit": 195,
          }
        }
      >
        <WithStyles(Modal)
          BackdropProps={
            Object {
              "transitionDuration": Object {
                "enter": 225,
                "exit": 195,
              },
            }
          }
          className="MuiDialog-root-7 MuiDialog-scrollPaper-8"
          disableBackdropClick={false}
          disableEscapeKeyDown={false}
          onClose={[Function]}
          open={false}
          role="dialog"
        >
          <Modal
            BackdropComponent={[Function]}
            BackdropProps={
              Object {
                "transitionDuration": Object {
                  "enter": 225,
                  "exit": 195,
                },
              }
            }
            className="MuiDialog-root-7 MuiDialog-scrollPaper-8"
            classes={
              Object {
                "hidden": "MuiModal-hidden-19",
                "root": "MuiModal-root-18",
              }
            }
            disableAutoFocus={false}
            disableBackdropClick={false}
            disableEnforceFocus={false}
            disableEscapeKeyDown={false}
            disablePortal={false}
            disableRestoreFocus={false}
            hideBackdrop={false}
            keepMounted={false}
            manager={
              ModalManager {
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "modals": Array [],
              }
            }
            onClose={[Function]}
            open={false}
            role="dialog"
          />
        </WithStyles(Modal)>
      </Dialog>
    </WithStyles(Dialog)>
  </Dialog>
</WithStyles(Dialog)>
`;
